File: src/main/java/org/example/handlers/ICollectionController.java
package org.example.handlers;

import org.example.collection.Person;
import org.example.exception.ArgumentError;
import org.example.exception.ScriptExecutionError;

import java.time.LocalDateTime;
import java.util.LinkedList;

public interface ICollectionController<T extends Comparable> {

    void add(T t);
    LocalDateTime getLastInitTime();
    String getCollectionType();
    public LinkedList<T> getCollection();
    int size();
    String getElements();
    String getElementsByAlbomCount(Integer integer);
    void updateElements(Integer id) throws ArgumentError, ScriptExecutionError;
    void removeElements(Integer index) throws ArgumentError;
    void clear();
    void removeElementByID(Integer id) throws ArgumentError;
    void shuffle();
    void reorder();
    void removeByFrontMan(Person person) throws ArgumentError;
    int countNumberOfParticipants(Long longs);
    void loadData();
    void saveData();
}

File: src/main/kotlin/handlers/ICollectionController.kt
package org.chousik.handlers

import org.chousik.collection.MusicBand
import org.chousik.collection.Person
import java.time.LocalDateTime
import java.util.*

interface ICollectionController<T : Comparable<*>> {

    fun add(t: MusicBand)

    val lastInitTime: LocalDateTime?

    val collectionType: String

    val collection: LinkedList<T>

    fun size(): Int

    val elements: String

    fun getElementsByAlbumCount(integer: Int): String

    fun updateElements(id: Int)

    fun removeElements(index: Int)

    fun clear()

    fun removeElementByID(id: Int)

    fun shuffle()

    fun reorder()

    fun removeByFrontMan(person: Person)

    fun countNumberOfParticipants(longs: Long): Int

    fun loadData()

    fun saveData(): Boolean
}